src/layer.js

```diff
@@ -148,6 +155,27 @@ export class MapLayer extends HTMLElement {
           this.opacity = newValue;
         }
       break;
+      case 'src':
```

Comment:
Now that you've added a src attributeChangedCallback case, you'll need to move the side effects of doing `<layer->.src = foo` into the attributeChangedCallback case, or make a _private method that encapsulates those side effects, and invoke it from the attributeChangedCallback case for src.  To me that means potentially what's in the src setter in this block

```javascript
if (this._layer) {
  let oldOpacity = this.opacity;
  // go through the same sequence as if the layer had been removed from
  // the DOM and re-attached with a new URL source.
  this.disconnectedCallback();
  var base = this.baseURI ? this.baseURI : document.baseURI;
  this._layer = M.mapMLLayer(val ? (new URL(val, base)).href: null, this);
  this._layer.on('extentload', this._onLayerExtentLoad, this);
  this._setUpEvents();
  if (this.parentNode) {
    this.connectedCallback();
  }
  this.opacity = oldOpacity;
}
```